mars/utils.py

```diff
@@ -544,34 +544,56 @@ def _is_fast_dtype(dtype):
         return sample_size * len(pd_obj) // max_samples
 
 
-def build_fetch_chunk(
-    chunk: ChunkType, input_chunk_keys: List[str] = None, **kwargs
+def build_fetch_shuffle(
+    chunk: ChunkType, n_reducers=None, shuffle_fetch_type=None
 ) -> ChunkType:
     from .core.operand import ShuffleProxy
+    from .core.operand import ShuffleFetchType
 
     chunk_op = chunk.op
+    assert isinstance(chunk_op, ShuffleProxy)
     params = chunk.params.copy()
+    n_mappers = len(chunk.inputs)
+    assert n_reducers > 0, n_reducers
+    # for shuffle nodes, we build FetchShuffle chunks
+    # to replace ShuffleProxy
+    source_keys, source_idxes, source_mappers = [], [], []
+    for pinp in chunk.inputs:
+        source_keys.append(pinp.key)
+        source_idxes.append(pinp.index)
```

Comment:
Why not put these lines to `elif shuffle_fetch_type == ShuffleFetchType.FETCH_BY_KEY`? It seems that when `shuffle_fetch_type == ShuffleFetchType.FETCH_BY_INDEX`, the `source_keys, source_idxes, source_mappers` are None.

Reply:
Fixed